File: fortran-tests/gemm-6.f90
subroutine gemm(m, n, k, alpha, A, B, C)
  integer :: m, n, k
  real, dimension(m, n) :: C
  real, dimension(m, k) :: A
  real, dimension(k, n) :: B
  real :: alpha
  real :: aa, bb

  do mm = 1, m
    do nn = 1, n
      do i = 1, k
        aa = A(mm, i)
        bb = B(nn, i)
        C(mm, nn) = C(mm, nn) + alpha * aa * bb
      end do
    end do
  end do
end subroutine

File: cplus-tests/gemm-6.cc
void basicSgemm(int m, int n, int k, float alpha, const float **A, const float **B, float **C)
{
  for (int mm = 0; mm < m; ++mm) {
    for (int nn = 0; nn < n; ++nn) {
      for (int i = 0; i < k; ++i) {
        float a = A[mm][i];
        float b = B[nn][i];
        C[mm][nn] += alpha * a * b;
      }
    }
  }
}
